File: capibara-website-scala-old/app/dtos/HeaderDetail.scala
package dtos
import models._
import play.api.libs.json.Json

object HeaderDetail {
  implicit val headerDetailFormatter = Json.format[HeaderDetail];
}

case class HeaderDetail(
    ref: String,
    name: String,
    summary: String,
    os_affinity: Seq[String],
    macros: Seq[Macro],
    enums: Seq[Enum],
    structs: Seq[Struct],
    typedefs: Seq[Typedef],
    functions: Seq[Function]
)

File: capibara-website/src/dtos/header_detail.rs
use rocket::serde::{Deserialize, Serialize};

use crate::models::_macro::Macro;
use crate::models::enumeration::Enumeration;
use crate::models::_struct::Struct;
use crate::models::typedef::Typedef;
use crate::models::function::Function;

#[derive(Serialize, Deserialize, Clone)]
pub(crate) struct HeaderDetail {
    #[serde(rename = "ref")]
    pub _ref: String,
    pub name: String,
    pub summary: String,
    pub os_affinity: Vec<String>,
    pub macros: Vec<Macro>,
    pub enums: Vec<Enumeration>,
    pub structs: Vec<Struct>,
    pub typedefs: Vec<Typedef>,
    pub functions: Vec<Function>
}